src/Bounds.cpp

```diff
@@ -111,6 +111,16 @@ class Bounds : public IRVisitor {
     Bounds(const Scope<Interval> *s, const FuncValueBounds &fb, bool const_bound)
         : func_bounds(fb), const_bound(const_bound) {
         scope.set_containing_scope(s);
+
+        // Find any points that are single_points but fail is_single_point due to
+        // pointer equality checks and replace with single_points.
+        for (auto item = s->cbegin(); item != s->cend(); ++item) {
+            // Updates to Scope<T> do not invalidate iterators.
```

Comment:
But we're not updating s, so we'd be fine anyway.

Reply:
Ah right - I will remove this comment.

Reply:
Removed in fc1d4e8